template/dictionary_types.njk

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2017 The Bacardi Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GEN_DICTIONARY_{{name | snakecase | upper}}_H_
+#define GEN_DICTIONARY_{{name | snakecase | upper}}_H_
+
+class {{name}} {
+public:
+{% for member in members %}
+  bool has{{member.name | pascalcase}}() const { return has_{{member.name}}_; }
+  {{member.type}} {{member.name}}() const {
+    return {{member.name}}_;
+  }
+  inline void set{{member.name | pascalcase}}({{member.type}});
+{% endfor %}
+private:
+{% for member in members %}
+  bool has_{{member.name}}_ = false;
```

Comment:
Always false?
It should be checked in runtime, so, it should not be generated I think.

Reply:
no, when javascript object will be converted to c++ object, it should be set true if that javascript object's attribute has value. actually that code is not exist yet. it should be implement.

Reply:
Yeah, I know but do we have to have the member variable to check whether it exists or not?
Is it just cache?

Reply:
um.. if there were boolean value in dictionary. and if it was passed as false value, it will be too hard to know that value came from javascript or it was just initial value.
so i think we have to know it was set or not.

Reply:
> um.. if there were boolean value in dictionary. and if it was passed as false value, it will be 
> too hard to know that value came from javascript or it was just initial value.
> so i think we have to know it was set or not.

The initial value you mentioned is a default value?

Reply:
No. it means initial value of native's members. and i think it will be used for default values of WebIDL.